src/App.js

```diff
@@ -1,16 +1,34 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const totalLikes = messages.reduce((total, entry) => {
+    total += entry.liked;
+    return total;
+  }, 0);
+
+  const setLike = (id) => {
+    setMessages((prev) => {
+      return prev.map((entry) => {
+        if (id === entry.id) {
+          return { ...entry, liked: !entry.liked };
```

Comment:
👍 Nice use of spreading and object shorthand to make a new copy fo the message being updated, while also toggling its liked status.